### PR TITLE
Incorporate F-RTO

### PR DESCRIPTION
**Pull request by ianswett**

### CODE REVIEW

File: draft-ietf-quic-recovery.md [hunk 1]
```diff
@@ -248,6 +248,10 @@ tlp_count:
 rto_count:
 : The number of times an rto has been sent without receiving an ack.
 
+largest_sent_before_rto:
+: The last packet number sent prior to the first transmission due to
```
Comment (janaiyengar) [inline draft-ietf-quic-recovery.md]: nit: change phrase to "prior to the first retransmission timeout"

File: draft-ietf-quic-recovery.md [hunk 2]
```diff
@@ -372,6 +377,12 @@ Pseudocode for OnPacketAcked follows:
 
 ~~~
    OnPacketAcked(acked_packet_number):
+     // If a packet sent prior to RTO was acked, then the RTO
+     // was spurious.  Otherwise, inform congestion control.
+     // Similar to the goal of F-RTO {{?RFC5682}}
```
Comment (janaiyengar) [inline draft-ietf-quic-recovery.md]: I would remove this last line from here and add it (now or later) to the text description.

File: draft-ietf-quic-recovery.md [hunk 3]
```diff
@@ -634,32 +650,48 @@ are detected lost.
 ~~~
    OnPacketsLost(lost_packets):
      largest_lost_packet = lost_packets.last()
+     // Start a new recovery epoch if the lost packet is larger
+     // than the end of the previous recovery epoch.
      if (end_of_recovery < largest_lost_packet.packet_number):
        end_of_recovery = largest_sent_packet
        congestion_window *= kLossReductionFactor
+       congestion_window = max(congestion_window, kMinimumWindow)
```
Comment (janaiyengar) [inline draft-ietf-quic-recovery.md]: nit: combine this line with the previous one:
congestion_window = 
    max(congestion_window * kLossReductionFactor, kMinimumWindow)

File: draft-ietf-quic-recovery.md [hunk 4]
```diff
@@ -634,32 +650,48 @@ are detected lost.
...
+sent immediately, and a time in the future if sending is pacing
+limited.
 
 ~~~
    TimeToSend(packet_size):
      if (bytes_in_flight + packet_size > congestion_window)
        return infinite
+     pacing_coefficient = 1.25
+     if (congestion_window < ssthresh)
+       pacing_coefficient = 2
```
Comment (janaiyengar) [inline draft-ietf-quic-recovery.md]: I don't think we need a pacing coeff of 2 here. I never understood why FQ did this, but I don't think we need to have this in the doc. In general I'm less certain that we want to describe the pacing impl as part of the congestion controller... I'm comfortable with definiing the TimeToSend interface without implementing it here. For now, I would at least remove the use of pacing_coeff here.

File: draft-ietf-quic-recovery.md (hunk 3 above)
Comment (ianswett) [inline draft-ietf-quic-recovery.md]: I tried, but it ends up taking up 3 lines, so I just kept it as is.

**Pull request merged**